### PR TITLE
Update UG and DG

### PR DESCRIPTION
**Pull request by songgthu**
UG:
- Add details for "Edit a class" feature

DG: 
- Add UML diagrams
- Describe the implementation of "Add a class" feature

### CODE REVIEW

**Review by revdrag — APPROVED** [review]
LGTM!

**Review by DonovanJJ — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -159,6 +159,50 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+#### Implementation
+
+Step 1. The user launches the application.
+
+Step 2. The user executes `add /c cs2103t` command to add a new class with the class name CS2103T. The `add /c` command calls `Parser#AddClassCommandParser` to retrieve the provided class name argument.
+
+Step 3. A new `Class` is created with the specified class name, an empty student list, an empty class note and empty class schedule.
+
+Step 4. The created `Class` is checked if the same class exists in `UniqueClassList`, then if not `Class` is added to the `UniqueClassList`, then update the `Storage` to save the current state of `EduTrack`.
```
Comment (seanpzk) [inline docs/DeveloperGuide.md]: Good to have the details and talks about the check. Perhaps, if you see fit, can shorten the sentence to : 
"The created `Class` is added to `UniqueClassList` if it does not already exists in `UniqueClassList`. The `Storage` then updates to save the current state of `EduTrack`."

**Review by seanpzk — APPROVED** [review]
LGTM, just a nitpick on DG 👍

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (songgthu) [reply docs/DeveloperGuide.md]: Thanks for your suggestion!

**Pull request merged**